### ISSUE TITLE
Nodes stuck on 100% cpu usage, making all the pods unresponsive. 

### ISSUE THREAD

**Issue by scm-ns** [kubernetes/kubernetes#68705]
**Is this a BUG REPORT or FEATURE REQUEST?**:

Serious Bug

**What happened**:
   Some of the nodes in my cluster reach 100% cpu usage and get stuck there. This has happened multiple times with different clusters that I have setup. It seems to be random, I let it run for the night and in the morning some nodes will be stuck at 100% cpu usage. Restarting the node and rejoining the cluster fixes the issues. Sometimes when I use the kubelet to create or destroy multiple nodes in succession, the nodes get stuck in the 100% cpu usage. 

**What you expected to happen**:
   The pods I am running don't use a lot of resources, so I am surprised that 100% cpu usage occurs. Even then, high cpu usage should not take the nodes out of the cluster. The nodes go into a NotReady state. 

**How to reproduce it (as minimally and precisely as possible)**:
Runs and few pods (~20) and tear them down. It happens every 2-3 days after starting a cluster. 

**Anything else we need to know?**:
This seems like something on the node getting stuck on a while loop. The server runs at 100%cpu usage for hours and the only way out is to force restart. sshing into the machine is not possible due to the 100%cpu usage. The nodes becomes unresponsive. 
I have a few databases running on the cluster, when this happens, the db volumes end up getting corrupted and I have to start using a new persistent volume

**Environment**:
- Kubernetes version :
Client Version: version.Info{Major:"1", Minor:"9", GitVersion:"v1.9.5", GitCommit:"f01a2bf98249a4db383560443a59bed0c13575df", GitTreeState:"clean", BuildDate:"2018-03-19T15:50:45Z", GoVersion:"go1.9.3", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"9", GitVersion:"v1.9.5", GitCommit:"f01a2bf98249a4db383560443a59bed0c13575df", GitTreeState:"clean", BuildDate:"2018-03-19T15:50:45Z", GoVersion:"go1.9.3", Compiler:"gc", Platform:"linux/amd64"}

- Cloud provider or hardware configuration:

Bare metal on digital ocean. 

- OS (e.g. from /etc/os-release):

ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 16.04.5 LTS"
VERSION_ID="16.04"
HOME_URL="http://www.ubuntu.com/"
SUPPORT_URL="http://help.ubuntu.com/"
BUG_REPORT_URL="http://bugs.launchpad.net/ubuntu/"
VERSION_CODENAME=xenial
UBUNTU_CODENAME=xenial

- Kernel (e.g. `uname -a`):

Linux 206.189.234.151 4.4.0-131-generic #157-Ubuntu SMP Thu Jul 12 15:51:36 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux

- Install tools:
  Kubespray

/sig scalability
/sig architecture
/sig scheduling
/kind bug

**Comment by sneko** [comment]
@scm-ns  @soumentrivedi I'm facing the same issue with a cluster managed by KOPS.

Do you have more information about this? As soon as the CPU usage is high my node is totally stuck 😢 

Thank you,

**Comment by JustDoItSascha** [comment]
Same Problem here. Just mongo databases without much traffic. Suddenly the CPU goes up to 100%.

**Comment by yaskinny** [comment]
@MetalPinguinInc Have You found the reason?

**Comment by MetalPinguinInc** [comment]
@yaskinny Yes, but actually no. I found some posts about QEMU/KVM possibly being the culprit. I am hosting my kubernetes nodes on QEMU/KVM guests. I have since updated qemu/kvm and have not run into the problem since. However, since I cannot reliably replicate the problem I do not know if it is actually fixed or if we have simply not run into the scenario that causes the issue. We have also considerably changed the deployments in the meantime. A sign that points to qemu/kvm being at fault though, is that I also had the same 100% CPU behaviour on a guest that had nothing to do with kubernetes. 

tl;dr I don't know, but update your qemu/kvm anyway.

**Comment by MetalPinguinInc** [comment]
@yaskinny Unfortunately this did not fix the issue. This morning we had another frozen node. Still no clue how to debug this since the logs are frozen as well and I can never find anything interesting in them after a reboot.

**Comment by MetalPinguinInc** [comment]
@yaskinny I have made some progress! I do not entirely understand the issue exactly or how to reproduce it, but I have found a forum post on the proxmox forums that describes this issue exactly. They seem to conclude there is an issue in the kernel that gets triggered when KVM checks if a pagefault (using is_page_fault_stale() ) in the guest is stale. This is caused by an integer overflow of the mmu_notifier_seq variable. I mostly put these words here so others can find the issue, because I don't actually understand how I can verify/reproduce this.

The good news is that this seems to be fixed (or will be fixed very soon) in the newest linux kernel. However, LTS releases of distros of course don't use the latest kernel, so the distro might need to do some backporting (at least that is what they imply in the proxmox thread).

I found this thread: https://forum.proxmox.com/threads/vms-freeze-with-100-cpu.127459/page-11#post-603296
And this mailing list response:  https://lore.kernel.org/stable/20230828101158.286647136@linuxfoundation.org/

Hopefully someone more familiar with kernel development can make complete sense of this.
